rpc: TestGRPCKeepaliveFailureFailsInflightRPCs failed

rpc.TestGRPCKeepaliveFailureFailsInflightRPCs [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10874260?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10874260?buildTab=artifacts#/) on master @ [c566d1f76f3a2be58670adbf0fd1c1d30f8a819f](https://github.com/cockroachdb/cockroach/commits/c566d1f76f3a2be58670adbf0fd1c1d30f8a819f):


```
=== RUN   TestGRPCKeepaliveFailureFailsInflightRPCs
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/a5e64047ddd0fa865a630ee2a4e95bc3/logTestGRPCKeepaliveFailureFailsInflightRPCs1048630171
    test_log_scope.go:81: use -show-logs to present logs inline
    context_test.go:1307: {cKeepalive:false sKeepalive:true partitionC2S:false partitionS2C:true expClose:false}: expected unclosed connection, found EOF
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/rpc.TestGRPCKeepaliveFailureFailsInflightRPCs.func3
          | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context_test.go:1297
        Wraps: (2) {cKeepalive:false sKeepalive:true partitionC2S:false partitionS2C:true expClose:false}
        Wraps: (3) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/rpc.grpcRunKeepaliveTestCase
          | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context_test.go:1526
          | github.com/cockroachdb/cockroach/pkg/rpc.TestGRPCKeepaliveFailureFailsInflightRPCs.func3
          | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context_test.go:1297
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_arm64.s:1172
        Wraps: (4) secondary error attachment
          | EOF
          | (1) EOF
          | Error types: (1) *errors.errorString
        Wraps: (5) expected unclosed connection, found EOF
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *secondary.withSecondaryError (5) *errutil.leafError
    context_test.go:1310: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a5e64047ddd0fa865a630ee2a4e95bc3/logTestGRPCKeepaliveFailureFailsInflightRPCs1048630171
--- FAIL: TestGRPCKeepaliveFailureFailsInflightRPCs (21.44s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGRPCKeepaliveFailureFailsInflightRPCs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-29649

rpc.TestGRPCKeepaliveFailureFailsInflightRPCs [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10884349?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/10884349?buildTab=artifacts#/) on master @ [6f0e7eaaae9b6854aa47d040bc3e9d4fc2527703](https://github.com/cockroachdb/cockroach/commits/6f0e7eaaae9b6854aa47d040bc3e9d4fc2527703):


```
=== RUN   TestGRPCKeepaliveFailureFailsInflightRPCs
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/a5e64047ddd0fa865a630ee2a4e95bc3/logTestGRPCKeepaliveFailureFailsInflightRPCs2700093030
    test_log_scope.go:81: use -show-logs to present logs inline
    context_test.go:1307: {cKeepalive:true sKeepalive:true partitionC2S:false partitionS2C:true expClose:true}: expected closed connection, found <nil>
        (1) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/rpc.TestGRPCKeepaliveFailureFailsInflightRPCs.func3
          | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context_test.go:1297
        Wraps: (2) {cKeepalive:true sKeepalive:true partitionC2S:false partitionS2C:true expClose:true}
        Wraps: (3) attached stack trace
          -- stack trace:
          | github.com/cockroachdb/cockroach/pkg/rpc.grpcRunKeepaliveTestCase
          | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context_test.go:1520
          | github.com/cockroachdb/cockroach/pkg/rpc.TestGRPCKeepaliveFailureFailsInflightRPCs.func3
          | 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context_test.go:1297
          | runtime.goexit
          | 	GOROOT/src/runtime/asm_arm64.s:1172
        Wraps: (4) expected closed connection, found <nil>
        Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.leafError
    context_test.go:1310: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/a5e64047ddd0fa865a630ee2a4e95bc3/logTestGRPCKeepaliveFailureFailsInflightRPCs2700093030
--- FAIL: TestGRPCKeepaliveFailureFailsInflightRPCs (22.43s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestGRPCKeepaliveFailureFailsInflightRPCs.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/rpc/context_test.go
+++ b/pkg/rpc/context_test.go
@@ -1234,6 +1234,7 @@ func TestRemoteOffsetUnhealthy(t *testing.T) {
 func TestGRPCKeepaliveFailureFailsInflightRPCs(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	skip.UnderShort(t, "Takes too long given https://github.com/grpc/grpc-go/pull/2642")
+	skip.UnderStress(t, "Under high parallelism (>100) on the host node, this test reuses ports and fails")
 
 	sc := log.Scope(t)
 	defer sc.Close(t)

